source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRMediaRenderer.mm

```diff
@@ -85,44 +85,44 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
     NSString *piikey = [NSString stringWithCString:[acoConfig getHostConfig]->GetMedia().playButton.c_str() encoding:[NSString defaultCStringEncoding]];
     UIImage *playIconImage = imageViewMap[piikey];
     UIImageView *playIconImageView = nil;
-    BOOL drawDefaultPlayIcon = YES;
+    BOOL hideDefaultPlayIcon = NO;
 
-    if(!playIconImage) {
-        playIconImageView  = [rootView getImageView:@"playIconImage"];
-    }else {
+    if (!playIconImage) {
+        NSNumber *number = [NSNumber numberWithUnsignedLongLong:(unsigned long long)elem.get()];
+        NSString *key = [NSString stringWithFormat:@"%@_%@", [number stringValue], @"playIcon" ];
```

Comment:
>stringWithFormat:@"%@_%@", [number stringValue], @"playIcon" [](start = 34, length = 60)

It isn't necessary to create an NSNumber here. You can just pass the number right into the string formatter: `... stringWithFormat:@"%llu_%@", (unsigned long long)elem.get(), ...`

You could actually also eliminate the other string format entry altogether if you wanted: `... stringWithFormat:@"%llu_playIcon", (unsigned long long)elem.get()]` Although if you prefer keeping the "playIcon" part separate for clarity, I have no objection.

Reply:
thanks!

---
In reply to: [299690929](https://github.com/microsoft/AdaptiveCards/pull/3105#discussion_r299690929) [](ancestors = 299690929)